myst_nb/cache.py

```diff
@@ -6,35 +6,52 @@
 from jupyter_cache.cache import main as cache
 from jupyter_cache.cli.utils import get_cache
 from jupyter_cache.executors import load_executor
+from nbclient import execute
 import datetime
 
 logger = logging.getLogger(__name__)
 
-def cached_execution_and_merge(file_path, ntbk):
+def add_notebook_outputs(file_path, ntbk, path_cache):
     """
+    Add outputs to a NotebookNode by pulling from cache and/or executing.
+
     Function to get the database instance and stage notebooks to it.
     Then cache the staged notebooks after execution, if not already present in the cache.
     The executed output is then merged with the original notebook. 
     """
-    cache_path = os.environ.get("JUPYTERCACHE", os.path.join(os.getcwd(), ".jupyter_cache"))
+    # If we have a jupyter_cache, see if there's a cache for this notebook
+    if path_cache:
+        try:
+            from jupyter_cache.cache.main import JupyterCacheBase
+        except ImportError:
+            logger.error(("Using caching functionality requires that "
+                          "jupyter_cache is installed. Install it first."))
 
-    db = get_cache(cache_path) #gets the db in the cache_path, if the db does not exist, then creates a db in this path
+        db = get_cache(path_cache)
 
-    # stage the notebook for execution
-    stage_record = db.stage_notebook_file(file_path)
+        # stage the notebook for execution
+        stage_record = db.stage_notebook_file(file_path)
 
-    # execute the notebook
-    execution_result = execute_nb(db, stage_record)
+        # execute the notebook
+        execution_result = execute_staged_nb(db, stage_record)
 
-    if len(execution_result['errored']) or len(execution_result['excepted']):
-        #handle case for errored and excpeted
-        pass
-    else:
-        pk, ntbk = db.merge_match_into_notebook(ntbk)
+        if len(execution_result['errored']) or len(execution_result['excepted']):
+            #handle case for errored and excpeted
+            pass
+        else:
+            pk, ntbk = db.merge_match_into_notebook(ntbk)
 
+        try:
```

Comment:
Should this try `try .. except` block be inside the `else` statement above https://github.com/AakashGfude/MyST-NB/pull/1/files#diff-e2066ea776dc9a4cfa8977efdc1df7abR41?

Reply:
yeah I think it could make sense to combine them somehow

Reply:
ok. I will merge the PR then. After that, we can play around.